bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/utils/tcl_esp-config.json

```diff
@@ -0,0 +1,80 @@
+{
+    "targets" : [
+        { "id": "esp32", "name": "ESP32" },
+        { "id": "esp32s2", "name": "ESP32-S2" },
+        { "id": "esp32s3", "name": "ESP32-S3" }
+    ],
+    "boards" : [
+        {
+            "name": "ESP-WROVER-KIT 3.3V",
+            "description": "ESP-WROVER-KIT with 3.3V ESP32-WROVER-B module",
+            "target": "esp32",
+            "config_files": ["boards/esp32-wrover-kit-3.3v.cfg"]
+        },
+        {
+            "name": "ESP-WROVER-KIT 1.8V",
+            "description": "ESP-WROVER-KIT with 1.8V ESP32-WROVER-B module",
+            "target": "esp32",
+            "config_files": ["boards/esp32-wrover-kit-1.8v.cfg"]
+        },
+        {
+            "name": "ESP32-ETHERNET-KIT",
+            "description": "ESP32-ETHERNET-KIT with ESP32-WROVER-E module",
+            "target": "esp32",
+            "config_files": ["board/esp32-ethernet-kit-3.3v.cfg"]
+        },
+        {
+            "name": "ESP32 module",
+            "description": "ESP32 used with ESP-PROG board",
+            "target": "esp32",
+            "config_files": [
+                "interface/ftdi/esp32_devkitjv1.cfg",
+                "target/esp32.cfg"
+            ]
+        },
+        {
+            "name": "ESP32-SOLO-1 module",
+            "description": "ESP32-SOLO-1 used with ESP-PROG board",
+            "target": "esp32",
+            "config_files": [
+                "interface/ftdi/esp32_devkitjv1.cfg",
+                "target/esp32-solo-1.cfg"
+            ]
+        },
+        {
+            "name": "ESP32-S2-KALUGA-1",
+            "description": "ESP32-S2-KALUGA-1 kit",
+            "target": "esp32s2",
+            "config_files": ["board/esp32s2-kaluga-1.cfg"]
+        },
+        {
+            "name": "ESP32-S2 module",
+            "description": "ESP32-S2 used with ESP-PROG board",
+            "target": "esp32s2",
+            "config_files": [
+                "interface/ftdi/esp32_devkitjv1.cfg",
+                "target/esp32s2.cfg"
+            ]
+        }
+    ],
+    "options": [
+        {
+            "name": "ESP_FLASH_VOLTAGE",
+            "values": ["default", "3.3", "1.8"],
```

Comment:
We can also implement them in the UI. By default, all HW options listed in options should have default value. It means that the value is chosen by OpenOCD itself. If the user alters the value special command should be added to OpenOCD command line before config files.

E.g. if user changes ESP_FLASH_VOLTAGE
to 3.3 OpenOCD should be started like openocd -c 'set ESP_FLASH_VOLTAGE
 3.3' -f board/esp32-ethernet-kit-3.3v.cfg

Reply:
Hi @kolipakakondal, this functionality was added in the new commit. 
Test cases for the new commit:

add JSON to path $OPENOCD_SCRIPTTS/esp-config.json. 

- By default, the target should match with the Launch target selected on the toolbar - passed
- select a board from the Board combo. Config options changed with board-specific configurations - passed
- change flash voltage. -c 'set ESP_FLASH_VOLTAGE {value}' is added before board configs, if the board is not selected,
config options are not changed - passed
- select a new target in the target combo. The list of boards in the board combo is updated - passed

delete JSON from the path $OPENOCD_SCRIPTTS/esp-config.json. 

- the new UI is not showing - passed